### PR TITLE
optimization yaml

### PR DESCRIPTION
**Pull request by ceci3**
optimization yaml

### CODE REVIEW

File: demo/auto_compression/detection/configs/ppyoloe_l_qat_dis.yaml [hunk 1]
```diff
@@ -8,14 +8,11 @@ Global:
...
-  distill_lambda: 1.0
-  distill_loss: l2_loss
-  distill_node_pair:
-  - teacher_concat_15.tmp_0
+  alpha: 1.0
+  loss: l2
+  node:
   - concat_15.tmp_0
-  - teacher_concat_14.tmp_0
   - concat_14.tmp_0
```
Comment (wanghaoshuang) [inline demo/auto_compression/detection/configs/ppyoloe_l_qat_dis.yaml:13]: 这里是指加两个蒸馏loss么？

File: demo/auto_compression/detection/configs/yolov3_mbv1_qat_dis.yaml [hunk 2]
```diff
@@ -7,16 +7,15 @@ Global:
...
-  distill_node_pair:
+  alpha: 1.0
+  loss: l2
+  node:
   - teacher_conv2d_84.tmp_0
   - conv2d_84.tmp_0
   - teacher_conv2d_85.tmp_0
   - conv2d_85.tmp_0
   - teacher_conv2d_86.tmp_0
   - conv2d_86.tmp_0
```
Comment (wanghaoshuang) [inline demo/auto_compression/detection/configs/yolov3_mbv1_qat_dis.yaml]: 这里的格式为什么没更新呢？是模型特殊么？

File: demo/auto_compression/hyperparameter_tutorial.md [hunk 3]
```diff
@@ -20,21 +20,19 @@ Quantization:
...
+    lambda: 1.0
+    # loss: 蒸馏loss算法；可输入多个loss，支持不同节点之间使用不同的loss算法
+    loss: l2
+    # node: 蒸馏节点，即某层输出的变量名称，可以选择：1. 要蒸馏的学生网络节点;
+    #                    2. 包含教师网络节点和对应的学生网络节点，
     #                    其中教师网络节点名称将在程序中自动添加 “teacher_” 前缀；
-    #                    可输入多个node_pair，支持多节点蒸馏
-    distill_node_pair:
-    - teacher_relu_30.tmp_0
+    #                    可输入多个node_pair，支持多节点蒸馏。
```
Comment (wanghaoshuang) [inline demo/auto_compression/hyperparameter_tutorial.md]: 注释也要更新下吧，最好详细解释下

File: demo/auto_compression/hyperparameter_tutorial.md [hunk 4]
```diff
@@ -43,16 +41,14 @@ Distillation:
...
 
-- 蒸馏loss目前支持的有：fsp_loss，l2_loss，soft_label_loss，也可自定义loss。具体定义和使用可参考[知识蒸馏API文档](https://paddleslim.readthedocs.io/zh_CN/latest/api_cn/static/dist/single_distiller_api.html)。
+- 蒸馏loss目前支持的有：fsp，l2，soft_label，也可自定义loss。具体定义和使用可参考[知识蒸馏API文档](https://paddleslim.readthedocs.io/zh_CN/latest/api_cn/static/dist/single_distiller_api.html)。
 
 
 #### 配置定制结构化稀疏策略
 
 结构化稀疏参数设置如下所示：
 ```yaml
 Prune:
```
Comment (wanghaoshuang) [inline demo/auto_compression/hyperparameter_tutorial.md]: 名字要不改成ChannelPrune？

File: demo/auto_compression/hyperparameter_tutorial.md [hunk 5]
```diff
@@ -61,9 +57,27 @@ Prune:
...
+  prune_params_name:
+  - conv1_weights
+```
+
+#### 配置定制针对Transformer结构的结构化剪枝策略
+
+针对Transformer结构的结构化剪枝参数设置如下所示：
+```yaml
+TransformerPrune:
+  # pruned_ratio: 裁剪比例
```
Comment (wanghaoshuang) [inline demo/auto_compression/hyperparameter_tutorial.md]: 注明是每层的剪枝比例，还是整体的剪枝比例

File: demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml [hunk 6]
```diff
@@ -1,9 +1,9 @@
-batch_size: 128
+batch_size: 2
```
Comment (wanghaoshuang) [inline demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml]: 这个是临时修改么？会影响精度么？

File: demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
 
 train_dataset:
   type: Dataset
-  dataset_root: data/humanseg
-  train_path: data/humanseg/train.txt
+  dataset_root: ./temp #data/humanseg
+  train_path: ./temp/train.txt # data/humanseg/train.txt
```
Comment (wanghaoshuang) [inline demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml]: 这些是为了debug引入的修改么？

File: demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml [hunk 8]
```diff
@@ -24,8 +24,8 @@ train_dataset:
 
 val_dataset:
   type: Dataset
-  dataset_root: data/humanseg
-  val_path: data/humanseg/val.txt
+  dataset_root: ./temp #data/humanseg
+  val_path: ./temp/val.txt #data/humanseg/val.txt
```
Comment (wanghaoshuang) [inline demo/auto_compression/semantic_segmentation/configs/pp_humanseg_lite.yaml]: 同上

File: paddleslim/auto_compression/compressor.py [hunk 9]
```diff
@@ -469,20 +482,20 @@ def single_strategy_compress(self, strategy, config, strategy_idx):
...
+            ###    _logger.info("metric of compressed model is: {}".format(metric))
+            ###    buf = 0.05
+            ###    if metric < (float(self.train_config.origin_metric) - buf) or \
+            ###            metric > (float(self.train_config.origin_metric) + buf):
+            ###        raise RuntimeError("target metric of pretrained model is {}, \
+            ###              but now is {}, Please check the format of evaluation dataset \
+            ###              or check the origin_metric in train_config"
+            ###                                                         .format(\
+            ###              self.train_config.origin_metric, metric))
+            ###    self.metric_before_compressed = metric
```
Comment (wanghaoshuang) [inline paddleslim/auto_compression/compressor.py]: 为了debug？

File: demo/auto_compression/detection/configs/ppyoloe_l_qat_dis.yaml [hunk 10]
```diff
@@ -8,14 +8,11 @@ Global:
...
-  merge_feed: true
   teacher_model_dir: ./ppyoloe_crn_l_300e_coco/
   teacher_model_filename: model.pdmodel
   teacher_params_filename: model.pdiparams
```
Comment (wanghaoshuang) [inline demo/auto_compression/detection/configs/ppyoloe_l_qat_dis.yaml]: 这三个可以设置为None或不设置么？直接用推理模型作为teacher model

**Review by wanghaoshuang — DISMISSED** [review]

**Review by wanghaoshuang — DISMISSED** [review]

**Review by yghstill — APPROVED** [review]
